apps/backend/api/src/common/enums/agent.ts

```diff
@@ -0,0 +1,18 @@
+export enum AgentType {
+  CAR_TOWING = 'CAR_TOWING',
+  DELIVERY = 'DELIVERY',
+}
+
+export enum AgentStatus {
+  ONLINE = 'ONLINE',
+  OFFLINE = 'OFFLINE',
```

Comment:
```suggestion
  OFFLINE = 'OFFLINE',
  
  WAITING_FOR_PICKUP = 'WAITING_FOR_PICKUP',  // The rider has accepted a delivery and is waiting for the pickup.
  EN_ROUTE_TO_PICKUP = 'EN_ROUTE_TO_PICKUP',  // The rider is traveling to the pickup location.
  PICKING_UP = 'PICKING_UP',  // The rider is at the pickup location and picking up the items.
  EN_ROUTE_TO_DELIVERY = 'EN_ROUTE_TO_DELIVERY',  // The rider is en route to the delivery location.
  DELIVERING = 'DELIVERING',  // The rider is at the delivery location completing the delivery.
  
```

Any thoughts on adding these status as well?

Reply:
I initially planned the status to simply indicate whether someone is online or not, but this is a great idea!

Reply:
So should we add these options into the enum?